databroker/core.py

```diff
@@ -1295,15 +1295,20 @@ def __init__(self,
             "Created %s for stream name %r",
             self.__class__.__name__,
             self._stream_name)
+        self._load()
```

Comment:
Does it work without this load?

Reply:
I first wrote the tests without this load (commit 877e63a) and they fail because `_load_partition` hasn't been called yet. Maybe the proper place to put this is in `_load_metadata`.

Reply:
Indeed. It looks like `_load` is a Catalog concept only. DataSources implement `_load_metadata`.

Reply:
...That's not quite right. But `_load_metadata` _is_ a thing and this seems like the right place to use it.

Reply:
I think we need to also change the other calls to _load in BlueskyEventStream to _load_metadata.

Reply:
there is another _load call here: https://github.com/bluesky/databroker/pull/511/files#diff-369696e385332d7139756bd5f292bd03R1337

Reply:
Digging through the base classes and intake-xarray a bit, I think the right thing to do for now is to invent our own method, called once at `__init__` time. Eventually we'll have to deal with reloading to check for new documents, something which is not currently handled, and at that point a separate method may turn out to be useful. It looks like `_load_metadata` is only meant to be called once, and `_load` is not in general a DataSource concept.